### DISCUSSION TITLE
Running UpSnap from a bridge?

### DISCUSSION THREAD

**Discussion by Jaxel** [seriousm4x/UpSnap#202] [Help wanted] [answered]
I try to keep all my Docker containers bridged, rather than on my host network. I'm having trouble configuring UpSnap for this.

This is my Docker compose:
```version: "3.8"
services:
  upsnap:
    image: ghcr.io/seriousm4x/upsnap:latest
    container_name: upsnap
    restart: unless-stopped
    network_mode: bridge
    ports:
      - 8090:8090
    volumes:
      - ./data:/app/pb_data
```

The problem I am having is that running a network scan in UpSnap doesn't find any of my devices. If I switch the network_mode to host, it can find my devices; but as I said, I'm trying to keep all my containers bridged.

What would I need to do, to get this working?

**Comment by seriousm4x** [comment]
I just did a little digging if thats possible.
The first answer that came up was this: https://stackoverflow.com/a/67018294

But I've found a blog post which might help you: https://www.devwithimagination.com/2020/06/15/homebridge-docker-and-wake-on-lan/
It's using a bridged network with a fixed ip address. Let me know if you can get it to work.

**Comment by MirisWisdom** [comment, marked as answer]
Steps which worked on my end:

1. Create an external network bridge: `docker network create biscuit-network`
2. In Docker Compose, link the UpSnap container to the created network bridge:
```yaml
services:
  upsnap:
    # ...

networks:
  default:
    name: biscuit-network
    external: true
```
3. On the Docker host, instruct `sysctl` with the following parameters:
```sh
sudo su

interface=$(docker network inspect biscuit-network --format '{{.Id}}')
interface=br-${interface:0:12}

sysctl net.ipv4.icmp_echo_ignore_broadcasts=0
sysctl net.ipv4.conf.all.bc_forwarding=1
sysctl net.ipv4.conf.${interface}.bc_forwarding=1
```

Tips:

You can persist the `sysctl` declarations, e.g. in `/etc/sysctl.d/99-upsnap.conf`.

To manually get the name of your bridge interface on the host, invoke `docker network ls` and refer to the `NETWORK ID` column; then, prepend the value with `br-`, e.g. `br-0946aec19ca1`.

You can also confirm it with:

```sh
ip a | grep 0946aec19ca1

# 6: br-0946aec19ca1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
#     link/ether 02:42:65:c0:2e:a2 brd ff:ff:ff:ff:ff:ff
#     inet 172.18.0.1/16 brd 172.18.255.255 scope global br-0946aec19ca1
#        valid_lft forever preferred_lft forever
#     inet6 fe80::42:65ff:fec0:2ea2/64 scope link proto kernel_ll 
#        valid_lft forever preferred_lft foreve
```
